hack/test-go.sh

```diff
@@ -109,87 +109,99 @@ if [[ ${#testcases[@]} -eq 0 ]]; then
 fi
 set -- "${testcases[@]+${testcases[@]}}"
 
-# TODO: this should probably be refactored to avoid code duplication with the
-# coverage version.
-if [[ $iterations -gt 1 ]]; then
-  if [[ $# -eq 0 ]]; then
-    set -- $(kube::test::find_dirs)
+runTests() {
+  echo "Running tests for api version ${KUBE_API_VERSION}"
+  # TODO: this should probably be refactored to avoid code duplication with the
+  # coverage version.
+  if [[ $iterations -gt 1 ]]; then
+    if [[ $# -eq 0 ]]; then
+      set -- $(kube::test::find_dirs)
+    fi
+    kube::log::status "Running ${iterations} times"
+    fails=0
+    for arg; do
+      trap 'exit 1' SIGINT
+      pkg=${KUBE_GO_PACKAGE}/${arg}
+      kube::log::status "${pkg}"
+      # keep going, even if there are failures
+      pass=0
+      count=0
+      for i in $(seq 1 ${iterations}); do
+        if go test "${goflags[@]:+${goflags[@]}}" \
+            ${KUBE_RACE} ${KUBE_TIMEOUT} "${pkg}"; then
+          pass=$((pass + 1))
+        else
+          fails=$((fails + 1))
+        fi
+        count=$((count + 1))
+      done 2>&1
+      kube::log::status "${pass} / ${count} passed"
+    done
+    if [[ ${fails} -gt 0 ]]; then
+      exit 1
+    else
+      exit 0
+    fi
   fi
-  kube::log::status "Running ${iterations} times"
-  fails=0
-  for arg; do
-    trap 'exit 1' SIGINT
-    pkg=${KUBE_GO_PACKAGE}/${arg}
-    kube::log::status "${pkg}"
-    # keep going, even if there are failures
-    pass=0
-    count=0
-    for i in $(seq 1 ${iterations}); do
-      if go test "${goflags[@]:+${goflags[@]}}" \
-          ${KUBE_RACE} ${KUBE_TIMEOUT} "${pkg}"; then
-        pass=$((pass + 1))
-      else
-        fails=$((fails + 1))
-      fi
-      count=$((count + 1))
-    done 2>&1
-    kube::log::status "${pass} / ${count} passed"
-  done
-  if [[ ${fails} -gt 0 ]]; then
-    exit 1
-  else
+
+  # If we're not collecting coverage, run all requested tests with one 'go test'
+  # command, which is much faster.
+  if [[ ! ${KUBE_COVER} =~ ^[yY]$ ]]; then
+    kube::log::status "Running unit tests without code coverage"
+    go test "${goflags[@]:+${goflags[@]}}" \
+      ${KUBE_RACE} ${KUBE_TIMEOUT} "${@+${@/#/${KUBE_GO_PACKAGE}/}}"
     exit 0
   fi
-fi
 
-# If we're not collecting coverage, run all requested tests with one 'go test'
-# command, which is much faster.
-if [[ ! ${KUBE_COVER} =~ ^[yY]$ ]]; then
-  kube::log::status "Running unit tests without code coverage"
-  go test "${goflags[@]:+${goflags[@]}}" \
-    ${KUBE_RACE} ${KUBE_TIMEOUT} "${@+${@/#/${KUBE_GO_PACKAGE}/}}"
-  exit 0
-fi
+  # Create coverage report directories.
+  cover_report_dir="/tmp/k8s_coverage/$(kube::util::sortable_date)"
```

Comment:
It would probably make sense for this to include the API version in the path (or some other way of differentiating meaningfully).